apps/api/internal/handler/article_test.go

```
@@ -0,0 +1,246 @@
+		w := httptest.NewRecorder()
+		h := NewArticleHandler(articleServiceFake{}, zennServiceFake{})
+		req := httptest.NewRequest(http.MethodGet, "/articles", nil)
+		h.ListArticles(w, req)
+		if w.Code != http.StatusOK {
+			t.Errorf("status = %d", w.Code)
+		}
+	})
+	t.Run("service error", func(t *testing.T) {
+		w := httptest.NewRecorder()
+		h := NewArticleHandler(articleServiceFake{err: sharedDependencyError()}, zennServiceFake{})
+		req := httptest.NewRequest(http.MethodGet, "/articles", nil)
```

golangci-lint (2.13.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)